src/Utilities/MoistThermodynamics/src/MoistThermodynamics.jl

```diff
@@ -35,247 +34,496 @@ export saturation_excess
+"""
+cv_m(ts::ThermodynamicState) =
+  cv_m(ts.q_tot, phase_partitioning_eq(ts)...)
 
 
 """
-    moist_gas_constants([q_tot=0, q_liq=0, q_ice=0])
+    (R_m, cp_m, cv_m, γ_m) = moist_gas_constants([q_tot=0, q_liq=0, q_ice=0])
+
+Wrapper to compute all gas constants at once, where optionally,
+
+ - `q_tot` total specific humidity
+ - `q_liq` liquid specific humidity
+ - `q_ice` ice specific humidity
+
+The function returns a tuple of
+ - `R_m` [`gas_constant_air`](@ref)
+ - `cp_m` [`cp_m`](@ref)
+ - `cv_m` [`cv_m`](@ref)
+ - `γ_m = cp_m/cv_m`
 
-Wrapper to return R_m, cv_m, cp_m, and gamma_m all at once
+Without the specific humidity arguments, the results
+are that of dry air.
 """
-function moist_gas_constants(q_tot=0, q_liq=0, q_ice=0)
+function moist_gas_constants(q_tot::DT=DT(0), q_liq::DT=DT(0), q_ice::DT=DT(0)) where DT
 
     R_gas  = gas_constant_air(q_tot, q_liq, q_ice)
     cp = cp_m(q_tot, q_liq, q_ice)
     cv = cv_m(q_tot, q_liq, q_ice)
-    gamma = cp/cv
+    γ = cp/cv
 
-    return (R_gas, cp, cv, gamma)
+    return (R_gas, cp, cv, γ)
 end
 
+"""
+    (R_m, cp_m, cv_m, γ_m) = moist_gas_constants(ts::ThermodynamicState)
+
+Wrapper to compute all gas constants at once, given a thermodynamic state `ts`.
+
+The function returns a tuple of
+ - `R_m` [`gas_constant_air`](@ref)
+ - `cp_m` [`cp_m`](@ref)
+ - `cv_m` [`cv_m`](@ref)
+ - `γ_m = cp_m/cv_m`
+
+"""
+moist_gas_constants(ts::ThermodynamicState) =
+  moist_gas_constants(ts.q_tot, phase_partitioning_eq(ts)...)
+
 """
     air_temperature(e_int[, q_tot=0, q_liq=0, q_ice=0])
 
-Return the air temperature, given the internal energy `e_int` per unit mass,
-and, optionally, the total specific humidity `q_tot`, the liquid specific humidity
-`q_liq`, and the ice specific humidity `q_ice`.
+The air temperature, where
+
+ - `e_int` internal energy per unit mass
+and, optionally,
+ - `q_tot` total specific humidity
+ - `q_liq` liquid specific humidity
+ - `q_ice` ice specific humidity
+
+Without the specific humidity arguments, it the results
+are that of dry air.
+"""
+function air_temperature(e_int::DT, q_tot::DT=DT(0), q_liq::DT=DT(0), q_ice::DT=DT(0)) where DT
+  T_0 +
+    (e_int - (q_tot - q_liq) * e_int_v0 + q_ice * (e_int_v0 + e_int_i0) ) /
+    cv_m(q_tot, q_liq, q_ice)
+end
+
 """
-function air_temperature(internal_energy, q_tot=0, q_liq=0, q_ice=0)
+    air_temperature_from_pottemp(θ, p[, q_tot=0, q_liq=0, q_ice=0])
+
+The air temperature, where
 
-    return T_0 +
-        ( internal_energy - (q_tot - q_liq) * e_int_v0 + q_ice * (e_int_v0 + e_int_i0) )/
-            cv_m(q_tot, q_liq, q_ice)
+ - `θ` potential temperature
+ - `p` pressure
+and, optionally,
+ - `q_tot` total specific humidity
+ - `q_liq` liquid specific humidity
+ - `q_ice` ice specific humidity
 
+Without the specific humidity arguments, it the results
+are that of dry air.
+"""
+function air_temperature_from_pottemp(θ::DT, p::DT, q_tot::DT=DT(0), q_liq::DT=DT(0), q_ice::DT=DT(0)) where DT
```

Comment:
I think you want `air_temperature_from_liquid_ice_pottemp` here, and solve the (linearized) liquid-ice potential temperature for `T`. This is what is needed for bracketing in saturation adjustment.